digdag-standards/src/main/java/io/digdag/standards/operator/jdbc/AbstractJdbcOperator.java

```diff
@@ -180,7 +179,11 @@ public TaskResult runTask()
                     logger.warn("Error during cleaning up status table. Ignoring.", ex);
                 }
 
-                return TaskResult.defaultBuilder(request).build();
+                // It's possible that the query already finished successfully, but this task don't know that
+                // if the previous connection with database was disconnected during waiting result.
+                // So setting `completedAt` in state is needed even if the task was already finished.
+                state.set(COMPLETED_AT, Instant.now());
+                throw TaskExecutionException.ofNextPolling(0, ConfigElement.copyOf(state));
```

Comment:
is this necessary? If this callback (setting completedAt stateParams) succeeds, another callback (making the task finished) should also succeed. So, I think that having this code doesn't improve risk significantly.